src/core/public/application/types.ts

```diff
@@ -0,0 +1,293 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Observable, Subject } from 'rxjs';
+
+import { Capabilities } from './capabilities';
+import { ChromeStart } from '../chrome';
+import { IContextProvider } from '../context';
+import { DocLinksStart } from '../doc_links';
+import { HttpStart } from '../http';
+import { I18nStart } from '../i18n';
+import { NotificationsStart } from '../notifications';
+import { OverlayStart } from '../overlays';
+import { PluginOpaqueId } from '../plugins';
+import { UiSettingsClientContract } from '../ui_settings';
+import { RecursiveReadonly } from '../../utils';
+
+/** @public */
+export interface AppBase {
+  id: string;
+
+  /**
+   * The title of the application.
+   */
+  title: string;
+
+  /**
+   * An ordinal used to sort nav links relative to one another for display.
+   */
+  order?: number;
```

Comment:
Why would we want `order` to be optional? If two or more apps don't specify an ordering then the ordering might not be guaranteed to be stable. I'm guessing if a non-unique order number is specified the ordering is affected by some underlying implementation detail like the order in which apps were registered, which in turn could be affected by the dependency graph?

Reply:
Correct, the ordering is currently stable by virtue of the plugin discovery's topological order algorithm. In practice, this means new apps tend to show up at the bottom of the list.

I'm not too keen on making plugin devs (mostly 3rd party ones) specify this field as it feels like just another detail to fill in that is unimportant.

Reply:
Yeah, I agree, from the perspective of 3rd party plugins it doesn't make sense to choose an ordering number.